### ISSUE TITLE
`--write-output` writing invalid JSON when using DBT templater

### ISSUE THREAD

**Issue by ttusing** [sqlfluff/sqlfluff#5054]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

See issue on `dbt-core`: https://github.com/dbt-labs/dbt-core/issues/8317 . This was also brought up in a similar form here: https://github.com/sqlfluff/sqlfluff-github-actions/issues/15#issuecomment-1030598058

The summary is that when using the DBT templater, certain output from DBT is written when using `--write-output`. This breaks a number of CI workflows, especially with `sqlfluff-github-actions` which uses the JSON output to annotate in Github workflows.

Is there somewhere that SQLFluff could be using  the `--quiet` flag when invoking DBT to avoid this output or otherwise avoiding this?

### Expected Behaviour

When running SQLFluff with `--write-output`, valid JSON is written and DBT logs are not included in the JSON file as plaintext.

### Observed Behaviour

JSON file contains this:
```
21:17:32  Registered adapter: snowflake=1.6.0
21:17:32  Unable to do partial parsing because saved manifest not found. Starting full parse.
[{"file": "myfile.sql", "line": 1, "start_column": 1, "end_column": 1, "title": "SQLFluff", "message": "LT05: Line is too long (110 > 100).", "annotation_level": "failure"}, {"file": "models/route66/common/myfile.sql", "line": 2, "start_column": 1, "end_column": 1, "title": "SQLFluff", "message": "LT05: Line is too long (103 > 100).", "annotation_level": "failure"}, {"file": "models/route66/common/dim_contract_pharmacies.sql", "line": 32, "start_column": 1, "end_column": 1, "title": "SQLFluff", "message": "LT02: Expected indent of 8 spaces.", "annotation_level": "failure"}]
```

### How to reproduce

Using DBT `1.5.4` and the DBT templater, run a common linting command, like:
```
sqlfluff lint --format github-annotation --annotation-level failure --nofail ${{ steps.get_files_to_lint.outputs.lintees }} > annotations.json
```
which is used frequently in workflows, or simply:
```
sqlfluff lint --format github-annotation --annotation-level failure --nofail {myfile.sql} > annotations.json
```

### Dialect

snowflake

### Version

`2.1.4`
DBT `1.5.4`

### Configuration

[sqlfluff]
templater = dbt
output_line_length = 100
max_line_length = 100
apply_dbt_builtins = True
dialect = snowflake

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by ttusing** [comment]
Might be fixed in DBT 1.6?

https://github.com/dbt-labs/dbt-core/issues/7872

**Comment by dbeatty10** [comment]
https://github.com/dbt-labs/dbt-core/issues/7872 is specifically for [programmatic invocations](https://docs.getdbt.com/reference/programmatic-invocations), which I don't think is currently in use by `sqlfulff`. It would be added in https://github.com/sqlfluff/sqlfluff/pull/4905 though.

Something else you could try: [`--write-output`](https://docs.sqlfluff.com/en/stable/cli.html#cmdoption-sqlfluff-lint-write-output) (#2625)

I haven't tried it myself, but appears to be something like this:
```shell
sqlfluff lint --format github-annotation --annotation-level failure --nofail ${{ steps.get_files_to_lint.outputs.lintees }} --write-output annotations.json
```

**Comment by alanmcruickshank** [comment]
Heya folks - if you're watching this issue, I've got a PR up which I think resolves this one. Check out #5070
